modules/storage/src/StockStorageAPI.php

```diff
@@ -21,180 +21,200 @@ public function createTransaction($variation_id, $location_id, $zone, $quantity,
     $data = isset($metadata['data']) ? $metadata['data'] : NULL;
 
     // Create a record.
-    // @todo - Deprecated replace with https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Database%21Connection.php/function/Connection%3A%3Ainsert/8
-    $io = db_insert('cs_inventory_transaction');
-    $io->fields([
-      'variation_id',
-      'qty',
-      'location_id',
-      'location_zone',
-      'unit_cost',
-      'transaction_time',
-      'transaction_type_id',
-      'related_tid',
-      'related_oid',
-      'related_uid',
-      'data',
-    ]);
-    $io->values([
-      $variation_id,
-      $quantity,
-      $location_id,
-      $zone,
-      $unit_cost,
-      time(),
-      $transaction_type_id,
-      $related_tid,
-      $related_oid,
-      $related_uid,
-      serialize($data),
-    ]);
-    return $io->execute();
+    $field_values = [
+      'variation_id' => $variation_id,
+      'qty' => $quantity,
+      'location_id' => $location_id,
+      'location_zone' => $zone,
+      'unit_cost' => $unit_cost,
+      'transaction_time' => time(),
+      'transaction_type_id' => $transaction_type_id,
+      'related_tid' => $related_tid,
+      'related_oid' => $related_oid,
+      'related_uid' => $related_uid,
+      'data' => serialize($data),
+    ];
+    $insert = \Drupal::database()->insert('cs_inventory_transaction')
+      ->fields(array_keys($field_values))
+      ->values(array_values($field_values))->execute();
+
+    return $insert;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function updateProductInventoryLocationLevel($location_id, $variation_id) {
-    // Get the location level & last transaction.
+  public function updateLocationStockLevel($location_id, $variation_id) {
+    $current_level = $this->getLocationStockLevel($location_id, $variation_id);
+    $last_update = $current_level['last_transaction'];
+    $latest_txn = $this->getLocationStockTransactionLatest($location_id, $variation_id);
+    $latest_sum = $this->getLocationStockTransactionSum($location_id, $variation_id, $last_update, $latest_txn);
+    $new_level = $current_level['qty'] + $latest_sum;
+
+    $this->setLocationStockLevel($location_id, $variation_id, $new_level, $latest_txn);
+  }
+
+  /**
+   * Gets stock level for a given location and variation.
+   *
+   * @param $location_id
+   *   Location id.
+   *
+   * @param $variation_id
+   *   Variation id.
+   *
+   * @return array
+   *   An array of 'qty' and 'last_transaction_id' values.
+   */
+  public function getLocationStockLevel($location_id, $variation_id) {
```

Comment:
This method already re-used by the method that determines qty for all locations. Also getLocationStockLevel used to set an initial record.  That didn't make sense and happens in a new setLocationStockLevel method.